docs/src/main/asciidoc/native-reference.adoc

```diff
@@ -0,0 +1,1382 @@
+
+One can see which Mandrel version was used to generate a binary by inspecting the binary as follows:
+
+[source,bash]
+----
+$ strings target/code-with-quarkus-1.0.0-SNAPSHOT-runner | grep GraalVM
+com.oracle.svm.core.VM=GraalVM 21.3.0.0-Final Java 11 Mandrel Distribution
+----
+
+=== How do I enable GC logging in native executables?
+
+Executing the native executable with `-XX:PrintFlags=` prints a list of flags that can be passed to native executables.
+For various levels of GC logging one may use:
+
+[source,bash]
+----
+$ ./target/code-with-quarkus-1.0.0-SNAPSHOT-runner -XX:PrintFlags=
+...
+  -XX:±PrintGC                                 Print summary GC information after each collection. Default: - (disabled).
+  -XX:±PrintGCSummary                          Print summary GC information after application main method returns. Default: - (disabled).
+  -XX:±PrintGCTimeStamps                       Print a time stamp at each collection, if +PrintGC or +VerboseGC. Default: - (disabled).
+  -XX:±PrintGCTimes                            Print the time for each of the phases of each collection, if +VerboseGC. Default: - (disabled).
+  -XX:±PrintHeapShape                          Print the shape of the heap before and after each collection, if +VerboseGC. Default: - (disabled).
+...
+  -XX:±TraceHeapChunks                         Trace heap chunks during collections, if +VerboseGC and +PrintHeapShape. Default: - (disabled).
+  -XX:±VerboseGC                               Print more information about the heap before and after each collection. Default: - (disabled).
+----
+
+=== Can I get a heap dump of a native executable? e.g. if it runs out of memory
+
+Unfortunately generating heap dumps in hprof format,
+which can be opened by tools such as VisualVM or Eclipse MAT,
+can only be achieved with
+https://www.graalvm.org/reference-manual/native-image/NativeImageHeapdump[GraalVM Enterprise Edition].
+Mandrel, which is based on the GraalVM Community Edition, does not have this capability.
+
+Although Mandrel can generate debug symbols and these contain a fair amount of information about object layouts,
+including what is a pointer field vs a primitive field, this information cannot be used as is to detect memory leaks or find dominator objects.
+This is because it has no idea what constitutes a root pointer nor how to recursively trace pointers from those roots.
+
+=== Can I create native executables for Windows?
+
+Quarkus applications can be converted into native executables for Windows,
+but this feature is not yet mature enough and so it has not yet been included in a RHBQ release.
+A Windows section will be included in this training once it gets included.
+
+[[macwindows]]
+=== Can I follow these examples if I’m running MacOS or Windows?
+
+The ideal environment for this training is Linux however, it’s still possible to follow all the steps running them within a Linux container environment in MacOS or Windows.
```

Comment:
Training?

Reply:
@galderz most (if not all) parts are working in macOS as well right? Maybe it's worth mentioning that as well.

Reply:
`gdb` debugging and profiling won't work on macOS native, but the rest yeah. I'll make a note.